workflow/executor/executor.go

```diff
@@ -722,15 +750,48 @@ func (we *WorkflowExecutor) AnnotateOutputs(ctx context.Context, logArt *wfv1.Ar
 	if !outputs.HasOutputs() {
 		return nil
 	}
-	log.Infof("Annotating pod with output")
-	outputBytes, err := json.Marshal(outputs)
+
+	if err := we.annotatePodWithOutputs(ctx, outputs); !apierr.IsForbidden(err) { // me were either successful (nil) or some other error
```

Comment:
I think this is left-over from the PoC and can be reverted

Reply:
2x

Reply:
I compared the master branch. code are just moved to function
![image](https://user-images.githubusercontent.com/33908564/123345359-bc459c80-d50a-11eb-848f-8f78da123e38.png)

Reply:
Am I missing anything?